### PR TITLE
[ROX-10609] Align component-cve relation datastore for postgres

### PR DESCRIPTION
**Pull request by md2119**
## Description

- Align component-cve datastore with postgres store
- Added searcher for component-cve datastore
- Unified composite ID format—primary keys separated by `#`—for  edge IDs, component IDs, cve IDs.
- Added a new SQL tag for id column to facilitate generation of `GetMany()`, etc. which are required in searchers.
- Removed some unused code

## Checklist
- [ ] Investigated and inspected CI test results
- [ ] Unit test and regression tests added
- ~Evaluated and added CHANGELOG entry if required~
- ~Determined and documented upgrade steps~
- ~Documented user facing changes~

If any of these don't apply, please comment below.

### CODE REVIEW

File: tools/generate-helpers/pg-table-bindings/store.go.tpl [hunk 1]
```diff
@@ -83,7 +86,9 @@ type Store interface {
 {{- if $singlePK }}
     GetIDs(ctx context.Context) ([]{{$singlePK.Type}}, error)
     GetMany(ctx context.Context, ids []{{$singlePK.Type}}) ([]*{{.Type}}, []int, error)
+{{- if not .JoinTable }}
```
Comment (md2119) [inline tools/generate-helpers/pg-table-bindings/store.go.tpl:89]: This was missed

File: pkg/search/postgres/utils.go [hunk 2]
```diff
@@ -0,0 +1,20 @@
+package postgres
+
+import (
+	"strings"
+)
+
+const (
+	// IDSeparator is the separator used in IDs constructed from multiple primary keys.
+	IDSeparator = "#"
```
Comment (md2119) [inline pkg/search/postgres/utils.go:9]: I used `#` for cves, etc. Let me know if it is not a good candidate.

**Review by viswajithiii — APPROVED** [review]
The whole ID thing needs some more thought IMO. But it's okay to merge this, we can refactor it in follow-ups if we think of something better (right now I can't)

File: central/componentcveedge/datastore/datastore_impl.go [hunk 3]
```diff
@@ -3,17 +3,30 @@ package datastore
...
 
+type componentCVEEdgePks struct {
+	componentID string
+	compName    string
+	compVersion string
+	compOS      string
+	cveID       string
+	cve         string
+	cveOS       string
+}
```
[the remark this answers is not in the document]
Comment (md2119) [reply central/componentcveedge/datastore/datastore_impl.go:28]: compID: (name#ver#os)
cveID: (cve#os)

File: central/componentcveedge/datastore/datastore_impl.go [hunk 4]
```diff
@@ -87,3 +105,20 @@ func (ds *datastoreImpl) filterReadable(ctx context.Context, ids []string) ([]st
 	})
 	return filteredIDs, err
 }
+
+func getPKs(id string) (componentCVEEdgePks, error) {
```
[the remark this answers is not in the document]
Comment (md2119) [reply central/componentcveedge/datastore/datastore_impl.go:109]: The stores for multiple pks are generated with pks as params. We will need to change the schema generator. I think we can come back to that later.

**Pull request merged**